app/core/models/ocelot/stanzas/SequenceStanza.scala

```diff
@@ -87,5 +86,24 @@ case class Sequence(text: Phrase,
 }
 
 object Sequence {
-  def apply(s: SequenceStanza, text: Phrase, options: Seq[Phrase]): Sequence = Sequence(text, s.next, options, s.label, s.stack)
+  def apply(s: SequenceStanza, text: Phrase, options: Seq[Phrase]): Sequence =
+    if (options.exists(o => exclusiveOptionRegex.findFirstMatchIn(o.english).nonEmpty))
+      ExclusiveSequence(text, s.next, options, s.label, s.stack)
+    else
+      NonExclusiveSequence(text, s.next, options, s.label, s.stack)
+}
+
+case class NonExclusiveSequence(text: Phrase,
+                                override val next: Seq[String],
+                                options: Seq[Phrase],
+                                label: Option[String],
+                                stack: Boolean) extends Sequence
+
+case class ExclusiveSequence(text: Phrase,
+                             override val next: Seq[String],
+                             options: Seq[Phrase],
+                             label: Option[String],
+                             stack: Boolean) extends Sequence {
+  lazy val exclusiveOptions: Seq[Phrase] = options.filter{p => exclusiveOptionRegex.findFirstMatchIn(p.english).nonEmpty}
```

Comment:
lazy on single call to partition could work here

Reply:
Done